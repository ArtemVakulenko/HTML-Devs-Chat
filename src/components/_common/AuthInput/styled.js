import styled from 'styled-components';

export const AuthInputContainer = styled.div`
  position: relative;
`;

export const AuthLabel = styled.p`
  padding: 3px;
  font-size: 18px;
  overflow: hidden;
`;
