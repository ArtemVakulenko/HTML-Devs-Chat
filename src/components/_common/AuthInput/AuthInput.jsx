import React from 'react';
import PropTypes from 'prop-types';
import { useTranslation } from 'react-i18next';
import Input from '../Input';
import { AuthInputContainer, AuthLabel } from './styled';

const AuthInput = ({ type, value, onChange }) => {
  const { t } = useTranslation();
  return (
    <AuthInputContainer>
      <AuthLabel>{t(type)}</AuthLabel>
      <Input
        value={value}
        type={type === 'password' || type === 'confirm' ? 'password' : 'text'}
        onChange={onChange}
        placeholder={`${type}Ipt`}
      />
    </AuthInputContainer>
  );
};

export default AuthInput;

AuthInput.propTypes = {
  type: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
};
