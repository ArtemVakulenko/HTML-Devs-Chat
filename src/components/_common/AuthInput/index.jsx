import { connect } from 'react-redux';
import { setAuthValue } from 'store/auth/authActions';
import { getAuthData } from 'store/auth/selectors';
import AuthInput from './AuthInput';

const mapStateToProps = (state, props) => ({
  value: getAuthData(state, props.type, props.page),
});
const mapDispatchToProps = (dispatch, props) => ({
  onChange: (value) => dispatch(setAuthValue(props.type, value, props.page)),
});

export default connect(mapStateToProps, mapDispatchToProps)(AuthInput);
