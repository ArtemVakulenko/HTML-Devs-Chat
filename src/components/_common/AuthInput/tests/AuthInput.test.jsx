import { shallow, mount } from 'enzyme';
import React from 'react';
import AuthInput from '../AuthInput';

jest.mock('react-i18next', () => ({
  useTranslation: jest.fn().mockImplementation(() => ({
    t: jest.fn(),
  })),
}));
jest.mock('../../../hooks', () => ({
  useTheme: jest.fn().mockImplementation(() => ({
    colors: {
      dark: {
        color: '',
        input: '',
        placeholder: '',
      },
    },
    theme: 'dark',
  })),
}));
describe('Title', () => {
  const props = {
    type: 'login',
    value: 'Alex',
    onChange: jest.fn(),
  };
  it('Should match snapshot', () => {
    const component = shallow(<AuthInput {...props} />);
    expect(component.html()).toMatchSnapshot();
  });
  it('Should have p', () => {
    const component = mount(<AuthInput {...props} />);
    expect(component.find('p')).toHaveLength(1);
  });
  it('Should have input', () => {
    const component = mount(<AuthInput {...props} />);
    expect(component.find('input')).toHaveLength(1);
  });
  it('Should have div', () => {
    const component = mount(<AuthInput {...props} />);
    expect(component.find('div')).toHaveLength(1);
  });
  it('Should have type password', () => {
    props.type = 'password';
    const component = mount(<AuthInput {...props} />);
    expect(component.find('div')).toHaveLength(1);
  });
});
