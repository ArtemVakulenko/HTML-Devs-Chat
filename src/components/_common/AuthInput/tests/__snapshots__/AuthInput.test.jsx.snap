// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title Should match snapshot 1`] = `"<div class=\\"styled__AuthInputContainer-iSEnWy kChWax\\"><p class=\\"styled__AuthLabel-jINklJ iykkVO\\"></p><input type=\\"text\\" value=\\"Alex\\" class=\\"styled__StInput-ifVcmW bBpwQJ\\"/></div>"`;
